src/persist/src/indexed/cache.rs

```diff
@@ -344,6 +350,37 @@ impl<B: Blob> BlobCache<B> {
         Ok(())
     }
 
+    fn check_meta_build_version(&self, meta: &ProtoMeta) -> Result<(), Error> {
+        // TODO: After ENCODING_VERSION is bumped to 8 or higher, this can be
```

Comment:
should we also rip out the code that deletes all persisted data on an encoding version < what was expected?

Reply:
> just to sanity check my understanding here -- that's because then any older versions will be covered by the encoding version check?

Yup, exactly. We'll only have written a proto with the field empty when encoding_version is 7. Once we're on encoding_version 8, we'll know that this field is always non-empty.

> should we also rip out the code that deletes all persisted data on an encoding version < what was expected?

Not quite yet. We'll keep that around until we actually commit to backward compatibility as an escape hatch. We also concretely plan to use it for changing the encoding of our data blobs to be some columnar thing (first your PR then a followup that'll use parquet or something) as well as changing the codec impls for Row + various errors that source persistence puts in K and V.

Reply:
sounds good.